### ISSUE TITLE
Bug: Focus remains on the menu bar after using Alt+Shift

### ISSUE THREAD

**Issue by trolley813** [getferdi/ferdi#1269]
**Bug description:**
When switching keyboard layout using `Alt+Shift` key combination (at least, in Linux and KDE), the focus remains on the menu bar (this does not happen in other applications, including Electron-based ones like Visual Studio Code). This is annoying since you cannot just continue typing (e.g.  when writing a message in several alphabets, like Latin and Cyrillic), because pressing letter keys without removing the focus (by pressing `Escape`) will trigger menu actions (in worst case, it would be quitting the entire application without saving the message draft, causing to retype it).

**Steps to reproduce:**
1. Install 2 or more keyboard layouts in Linux (at least on KDE, but may affect other DEs) and assign the `Alt+Shift` keyboard combination to layout switching.
2. Open Ferdi, start typing a message in a service (e.g. Telegram or WhatsApp) and switch the keyboard layout (as usual, pressing `Alt` first, then pressing `Shift` while holding `Alt`, and then release both keys).
3. After releasing `Alt` key, the focus remains on the menu bar.

**Expected behavior**
After step 2, the focus should return to e.g. the message typing area.

**Environment:**

- Operating System: Linux (Fedora 33 with KDE), kernel 5.11.11
- Ferdi Version: 5.5.0
- Server: Without using an account

- Debug information: None applicable

**Comment by korry73** [comment]
I have the same problem, it is annoying, could we expect a fix?

**Comment by kris7t** [comment]
Version 5.5.0 is extremely old now. Unfortunately, nightly builds are still broken in CI, but you should try building Ferdi from source and check whether it fixes this. I can't reproduce with the latest nightly (which uses Electron 12, so it's likely this was fixed in Electron somewhat recently).

If it's still broken with KDE (I'm on sway, so a different window manager, plus Wayland instead of X11), it's possible this is an upstream Electron limitation, but vscode implements their own custom menubar (integrated with the title bar on e.g., on windows) without this limitation. In that case, implementing our own menubar widget is probably out of scope, but contributions are still welcome, of course.

**Comment by korry73** [comment]
Well, I should have definitely mentioned it, I am on 5.6.0 nightly 15 and the issue occurs. The rest is similar to trolley813's setup - latest KDE (Kubuntu).

I can try to compile from source in a VM and test.

**Comment by kris7t** [comment]
Nightly 15 is still very old and use Electron 9 vs git `develop` with Electron 13. Unfortunately, nightly builds in CI are broken atm, so you'll have to build from source.

**Comment by vraravam** [comment]
Or, if you are on macos, you can find the latest versions (built from source) over [here](https://drive.google.com/drive/folders/1mJmKbhpVHdb066IHlrdYiD8TG9wlmuBv)

**Comment by kris7t** [comment]
> at least, in Linux and KDE

I guess the macos builds won't be that useful for you, then. I'm currently fixing some problems with building on Linux with Electron 13 (it's so fresh it isn't even on https://www.electronjs.org/, but the final 13.0.0 package is uploaded to npmjs at last  😀).

**Comment by MurzNN** [comment]
This issue is fixed on Electron side in 13.1.0 version, will wait when Ferdi updates to this version.

**Comment by vraravam** [comment]
@MurzNN - Ferdi is already on electron 13.1.4 - please use the nightlies from [here](http://github.com/vraravam/nightlies/releases)

**Comment by korry73** [comment]
It seems the issue is fixed in Ferdi nightly 68

**Comment by mhatvan** [comment]
Closing due to fix with electron 13.1.4
